sor-api/src/main/java/com/bazaarvoice/emodb/sor/condition/impl/AndConditionImpl.java

```diff
@@ -40,13 +55,39 @@ public void appendTo(Appendable buf) throws IOException {
         buf.append(")");
     }
 
+    /**
+     * The worst case total weight of an "and" is the sum of the weights of all contained conditions.
+     */
+    @Override
+    public int weight() {
+        return _conditions.stream().mapToInt(Condition::weight).sum();
+    }
+
     @Override
     public boolean equals(Object o) {
-        return this == o || (o instanceof AndCondition) && _conditions.equals(((AndCondition) o).getConditions());
+        return this == o || (o instanceof AndCondition) && conditionsEqual(((AndCondition) o).getConditions());
+    }
+
+    /**
+     * The order of the conditions is irrelevant, just check the set is the same.
+     */
+    private boolean conditionsEqual(Collection<Condition> conditions) {
+        if (conditions.size() != _conditions.size()) {
+            return false;
+        }
+        List<Condition> unvalidatedConditions = new ArrayList<>(conditions);
+        for (Condition condition : _conditions) {
+            if (!unvalidatedConditions.remove(condition)) {
+                return false;
+            }
+        }
+        return true;
     }
 
     @Override
     public int hashCode() {
-        return 11213 ^ _conditions.hashCode();
+        // Order of the underlying collection of values is irrelevant, so sum the individual object hashes
+        // so that order does not affect the computed hash.
+        return 11213 ^ _conditions.stream().mapToInt(Object::hashCode).sum();
```

Comment:
How did we come to 11213?

Reply:
It's randomly generated by IntelliJ to create a unique hash code in case multiple similar documents have similar attributes.  Nowadays we'd just use `Objects.hash(Object...)`, but in this case it's different enough that I kept the general calculation as-is.

Reply:
FWIW it would be equally valid to simply `return _conditions...sum()` without the exponent, especially if a) this call will only ever be hashed with other instances of the same class and b) this only comes up in unit tests, which it does.  If you feel it improves comprehension I can change it.